pcsx2/CDVD/Ps1CD.cpp

```diff
@@ -365,14 +365,28 @@ void cdrInterrupt()
 
 		case CdlGetlocP:
 			SetResultSize(8);
-			cdr.Result[0] = 1;
-			cdr.Result[1] = 1;
-			cdr.Result[2] = cdr.Prev[0];
-			cdr.Result[3] = itob((btoi(cdr.Prev[1])) - 2);
-			cdr.Result[4] = cdr.Prev[2];
-			cdr.Result[5] = cdr.Prev[0];
-			cdr.Result[6] = cdr.Prev[1];
-			cdr.Result[7] = cdr.Prev[2];
+			cdr.subQ = new cdvdSubQ();
```

Comment:
This is a memory leak. I'd also question the logic here - you should have subq for any sector, including those not in the data/audio tracks (lead-in/lead-out).

Reply:
Is what I force-pushed any better?

Reply:
Not really. A heap allocation here is completely pointless, when you could just store it on the stack instead.

And what happens if readSubQ does error (e.g. due to using a physical disc and a read error)? The result is left with undefined and/or junk data?

Reply:
I put it on the stack, and if the main subQ read doesn't work, it'll fake it instead

Reply:
Wasn't that the entire point of this PR?

Reply:
Yeah, I just wasn't sure if the original logic for the faked subQ was the best so I tried to ax it. In hindsight, that was a stupid decision :sweat_smile: